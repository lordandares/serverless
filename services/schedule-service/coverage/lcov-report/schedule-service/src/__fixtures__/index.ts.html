<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for schedule-service/src/__fixtures__/index.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">schedule-service/src/__fixtures__</a> index.ts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145</td><td class="line-coverage quiet"><span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const scheduleDocument = {
  applicationId: 'application1',
  createdAt: '2019-10-13T00:00:00.000Z',
  createdBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
  data: {
    areas: ['area1'],
    enabled: true,
    locations: ['location1'],
    name: 'Schedule One',
    included: {},
    serviceHours: {
      hours: [
        {
          description: 'MON 11:00 - TUE 00:00',
          end: 86400000,
          start: 39600000,
          type: 'DEFAULT',
        },
      ],
      timezone: 'Australia/Melbourne',
    },
    strategy: {
      options: {
        duration: {
          unit: 'minute',
          value: 15,
        },
        frequency: {
          unit: 'minute',
          value: 15,
        },
      },
      type: 'stopwatch',
    },
    type: 'visit',
  },
  endAt: '2019-10-1500:00:00.000Z',
  groupType: 'schedule',
  pk: 'application1-schedule',
  scheduleId: 'schedule1',
  sk: '2019-10-13T00:00:00.000Z-schedule',
  startAt: '2019-10-1400:00:00.000Z',
  updatedAt: '2019-10-13T01:00:00.000Z',
  updatedBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
}
&nbsp;
export const schedulePayload = {
  areas: ['area1'],
  enabled: true,
  endAt: '2019-10-1500:00:00.000Z',
  locations: ['location1'],
  name: 'Schedule One',
  serviceHours: {
    hours: [
      {
        description: 'MON 11:00 - TUE 00:00',
        end: 86400000,
        start: 39600000,
        type: 'DEFAULT',
      },
    ],
    timezone: 'Australia/Melbourne',
  },
  startAt: '2019-10-1400:00:00.000Z',
  strategy: {
    options: {
      duration: {
        unit: 'minute',
        value: 15,
      },
      frequency: {
        unit: 'minute',
        value: 15,
      },
    },
    type: 'stopwatch',
  },
  type: 'visit',
}
&nbsp;
export const occurrenceDocument = {
  applicationId: 'application1',
  createdAt: '2019-01-01T00:00:00.000Z',
  createdBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
  endAt: '2019-02-02T00:00:00.000Z',
  groupType: 'occurrence',
  location: 'location1',
  occurrenceId: 'occurrence1',
  pk: 'application1-occurrence',
  scheduleId: 'schedule1',
  sk: '2019-01-01T00:00-occurrence',
  startAt: '2019-02-01T00:00:00.000Z',
  type: 'task',
  updatedBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
}
&nbsp;
export const areaDocument = {
  _id: 'location1',
  application: '56789671d4c6488b128c7f9c',
  name: 'WRNMMC Facility',
  altName: 'WRNMMC Facility',
  type: 'location',
  tags: [],
  address: {
    street: '8901 Rockville Pike',
    street2: '',
    city: 'Bethesda',
    state: 'MD',
    postal: '20889',
    country: 'US',
  },
  keywords: ['hospital'],
  areaSize: 536559.55514508,
  serviceHours: {
    hours: [
      {
        description: 'MON 00:00 - TUE 00:00',
        duration: 86400000,
        end: 86400000,
        start: 0,
        type: 'DEFAULT',
      },
    ],
    timezone: 'America/New_York',
  },
  deleted: false,
  timezone: 'America/New_York',
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 25 2019 18:46:45 GMT+0000 (Coordinated Universal Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
