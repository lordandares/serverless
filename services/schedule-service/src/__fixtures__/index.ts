import { Context } from 'aws-lambda'

export const areaDocument = {
  _id: 'location1',
  address: {
    city: 'Bethesda',
    country: 'US',
    postal: '20889',
    state: 'MD',
    street: '8901 Rockville Pike',
    street2: '',
  },
  altName: 'WRNMMC Facility',
  application: '56789671d4c6488b128c7f9c',
  areaSize: 536559.55514508,
  deleted: false,
  floorsRef: [0],
  groupType: 'location',
  keywords: ['hospital'],
  name: 'WRNMMC Facility',
  serviceHours: {
    hours: [
      {
        description: 'MON 00:00 - TUE 00:00',
        duration: 86400000,
        end: 86400000,
        start: 0,
        type: 'DEFAULT',
      },
    ],
    timezone: 'America/New_York',
  },
  tags: [],
  timezone: 'America/New_York',
}

export const occurrenceDocument = {
  applicationId: 'application1',
  createdAt: '2019-01-01T00:00:00.000Z',
  createdBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
  data: {
    occurrenceInterval: [1, 2],
    scheduleName: 'Schedule 1',
    serviceInterval: [1, 2],
    timezone: 'Australia/Melbourne',
    type: 'visit',
  },
  endAt: '2019-02-01T23:59:59.999Z',
  groupType: 'occurrence',
  locationId: '5d163137b8b3b7000127edd1',
  location_endAt_occurrenceId: 'location1-2019-02-02T00:00:00.000Z-occurrence1',
  occurrenceId: 'occurrence1',
  pk: 'application1-occurrence',
  scheduleId: 'schedule1',
  sk: '2019-01-01T00:00-occurrence1',
  startAt: '2019-02-01T00:00:00.000Z',
  status: 'pending',
  updatedBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
}

export const occurrenceTimerDocument = {
  pk: `timer-2019-01-01T00:00`,
  sk: `timer#application1-occurrence#2019-01-01T00:00-occurrence`,
  endAt: '2019-01-01T23:59:59.999Z',
  groupType: `occurrence-timer`,
  scheduleId: 'schedule1',
  targetArn: 'someArn',
}

export const scheduleDocument = {
  applicationId: 'application1',
  createdAt: '2019-10-13T00:00:00.000Z',
  createdBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
  data: {
    areas: ['area1'],
    enabled: true,
    included: {},
    locations: ['location1'],
    name: 'Schedule One',
    serviceHours: {
      hours: [
        {
          description: 'MON 11:00 - TUE 00:00',
          end: 86400000,
          start: 39600000,
          type: 'DEFAULT',
        },
      ],
      timezone: 'Australia/Melbourne',
    },
    strategy: {
      options: {
        duration: {
          unit: 'minute',
          value: 15,
        },
        frequency: {
          unit: 'minute',
          value: 15,
        },
      },
      type: 'stopwatch',
    },
    type: 'visit',
  },
  endAt: '2019-10-1500:00:00.000Z',
  groupType: 'schedule',
  pk: 'application1-schedule',
  scheduleId: 'schedule1',
  sk: '2019-10-13T00:00:00.000Z-schedule1',
  startAt: '2019-10-1400:00:00.000Z',
  updatedAt: '2019-10-13T01:00:00.000Z',
  updatedBy: {
    id: 'user1',
    label: 'Unknown User',
    type: 'user',
  },
}

export const scheduleLocationDocument = {
  applicationId: 'application1',
  createdAt: '2019-10-13T00:00:00.000Z',
  createdBy: {
    id: '123456789',
    label: 'label',
    type: 'system',
  },
  data: {
    name: 'Schedule One',
    serviceHours: {
      hours: [
        {
          description: 'MON 11:00 - TUE 00:00',
          end: 86400000,
          start: 39600000,
          type: 'DEFAULT',
        },
      ],
      timezone: 'Australia/Melbourne',
    },
  },
  groupType: 'location',
  pk: 'application1-location',
  schedules: ['schedule1'],
  sk: 'location1',
  updatedAt: '2019-10-13T01:00:00.000Z',
  updatedBy: {
    id: '123456789',
    label: 'label',
    type: 'system',
  },
}

export const schedulePayload = {
  areas: ['area1'],
  enabled: true,
  endAt: '2019-10-1500:00:00.000Z',
  locations: ['location1'],
  name: 'Schedule One',
  serviceHours: {
    hours: [
      {
        description: 'MON 11:00 - TUE 00:00',
        end: 86400000,
        start: 39600000,
        type: 'DEFAULT',
      },
    ],
    timezone: 'Australia/Melbourne',
  },
  startAt: '2019-10-1400:00:00.000Z',
  strategy: {
    options: {
      duration: {
        unit: 'minute',
        value: 15,
      },
      frequency: {
        unit: 'minute',
        value: 15,
      },
    },
    type: 'stopwatch',
  },
  type: 'visit',
}

export const rulePayload = {
  applicationId: 'application-id-1111',
  data: {
    occurrenceInterval: [1580761454998, 1580762054997],
  },
  hook: {
    endpoint: 'sns:endpoint:uri',
    type: 'sns',
  },
  locationId: 'location-id-11111',
  occurrenceId: 'occurrence1',
  pk: 'application1-occurrence',
  scheduleId: 'schedule1',
  sk: '2019-06-02T00:00:00.000Z-occurrence1',
  startAt: '2019-02-01T00:00:00.000Z',
  type: 'visit',
}

export const rulePatternDocument = {
  createdAt: '2019-06-02T00:00:00.000Z',
  createdBy: {
    id: 'created-by-id-11111',
    label: 'created-by-label-1111',
    type: 'created-by-type-11111',
  },
  groupType: 'rule-pattern',
  matches: {
    occurrence1: {
      pk: 'application1-occurrence',
      sk: '2019-06-01T00:00:00.000Z-occurrence1',
    },
  },
  pk: 'rule-pattern-application-id-11111-location-id-11111',
  sk: 'visit',
  updatedAt: '2019-06-02T00:00:00.000Z',
}

export const context: Context = {
  awsRequestId: 'aws-request-id',
  callbackWaitsForEmptyEventLoop: false,
  clientContext: undefined,
  done: jest.fn(),
  fail: jest.fn(),
  functionName: 'function-name',
  functionVersion: 'function-version',
  getRemainingTimeInMillis: jest.fn(),
  identity: undefined,
  invokedFunctionArn: 'invoked-function-arn',
  logGroupName: 'log-group-name',
  logStreamName: 'log-stream-name',
  memoryLimitInMB: 0,
  succeed: jest.fn(),
}
