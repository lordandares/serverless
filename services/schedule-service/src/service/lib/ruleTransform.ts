import { schemas } from '@lighthouse/serverless-common'
import * as moment from 'moment'
import { v4 as uuid } from 'uuid'

export interface PayloadToRulePatternDocument {
  actor: schemas.ActorSchema
  existingDocument: schemas.RulePatternDocumentSchema | undefined
  payload: {
    applicationId: string
    locationId: string
    occurrenceId: string
    pk: string
    sk: string
    type: string
  }
}
export function payloadToRulePatternDocument({
  actor,
  existingDocument,
  payload,
}: PayloadToRulePatternDocument): schemas.RulePatternDocumentSchema {
  const { applicationId, locationId, occurrenceId, pk, sk, type } = payload

  const now: string = moment
    .utc()
    .toDate()
    .toISOString()

  const matchValues = { pk, sk }

  const matches: schemas.MatchSchema =
    existingDocument && existingDocument.matches
      ? { ...existingDocument.matches, [occurrenceId]: matchValues }
      : { [occurrenceId]: matchValues }

  const document: schemas.RulePatternDocumentSchema = {
    createdAt: (existingDocument && existingDocument.createdAt) || now,
    createdBy: (existingDocument && existingDocument.createdBy) || actor,
    groupType: 'rule-pattern',
    matches,
    pk: `rule-pattern-${applicationId}-${locationId}`,
    sk: type,
    updatedAt: now,
    updatedBy: actor,
  }

  return document
}
