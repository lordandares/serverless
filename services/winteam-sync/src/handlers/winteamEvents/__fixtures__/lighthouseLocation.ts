const lighthouseLocation = {
  _id: 'objectId',
  name: 'Powell Motors',
  type: 'location',
  address: {
    street: 'Center',
    street2: '7855 So. River Pkwy',
    city: 'San Diego',
    state: 'CA',
    postalCode: '852840000',
    country: 'US',
  },
  geometry: {
    type: 'Polygon',
    coordinates: [
      [
        [39.984, -75.34310067963625],
        [39.98365162727836, -75.34310500985033],
        [39.9833066089773, -75.34311795879778],
        [39.98296826723548, -75.3431394017952],
        [39.982639859937834, -75.34316913237075],
        [39.98232454936074, -75.34320686425103],
        [39.98202537173541, -75.3432522341163],
        [39.98174520802207, -75.34330480509715],
        [39.98148675617558, -75.34336407097913],
        [39.98125250516942, -75.34342946107526],
        [39.9810447110278, -75.34350034571895],
        [39.98086537509659, -75.34357604232524],
        [39.980716224762624, -75.34365582196158],
        [39.980598696807064, -75.34373891636542],
        [39.98051392355378, -75.34382452534051],
        [39.98046272194656, -75.34391182446154],
        [39.98044558566106, -75.3439999730124],
        [39.98046268032836, -75.34408812208183],
        [39.98051384191675, -75.3441754227386],
        [39.980598578288465, -75.34426103420762],
        [39.98071607391706, -75.3443441319677],
        [39.98086519772097, -75.34442391569367],
        [39.98104451393857, -75.34449961696573],
        [39.981252295940614, -75.34457050667213],
        [39.98148654284773, -75.34463590203326],
        [39.98174499879326, -75.34469517318026],
        [39.98202517464618, -75.34474774922379],
        [39.98232437198512, -75.34479312375487],
        [39.982639709092275, -75.34483085972474],
        [39.98296814871688, -75.34486059365653],
        [39.98330652734027, -75.34488203914788],
        [39.98365158566017, -75.34489498963106],
        [39.984, -75.34489932036371],
        [39.98434841433984, -75.34489498963106],
        [39.984693472659735, -75.34488203914788],
        [39.985031851283125, -75.34486059365653],
        [39.98536029090773, -75.34483085972474],
        [39.98567562801488, -75.34479312375487],
        [39.98597482535382, -75.34474774922379],
        [39.98625500120675, -75.34469517318026],
        [39.98651345715227, -75.34463590203326],
        [39.98674770405939, -75.34457050667213],
        [39.98695548606143, -75.34449961696573],
        [39.98713480227903, -75.34442391569367],
        [39.987283926082945, -75.3443441319677],
        [39.987401421711546, -75.34426103420762],
        [39.98748615808325, -75.3441754227386],
        [39.987537319671645, -75.34408812208183],
        [39.987554414338945, -75.3439999730124],
        [39.98753727805345, -75.34391182446154],
        [39.987486076446224, -75.34382452534051],
        [39.98740130319294, -75.34373891636542],
        [39.98728377523738, -75.34365582196158],
        [39.98713462490341, -75.34357604232524],
        [39.98695528897221, -75.34350034571895],
        [39.98674749483058, -75.34342946107526],
        [39.986513243824426, -75.34336407097913],
        [39.98625479197794, -75.34330480509715],
        [39.98597462826459, -75.3432522341163],
        [39.98567545063927, -75.34320686425103],
        [39.98536014006217, -75.34316913237075],
        [39.98503173276453, -75.3431394017952],
        [39.9846933910227, -75.34311795879778],
        [39.984348372721634, -75.34310500985033],
        [39.984, -75.34310067963625],
      ],
    ],
  },
  plugins: {
    winteam: {
      options: {
        jobId: '947049c7-e118-4ee9-8e2e-c956285917f1',
        jobNumber: '50012',
      },
    },
  },
  timezone: 'America/Los_Angeles',
}

export default lighthouseLocation
