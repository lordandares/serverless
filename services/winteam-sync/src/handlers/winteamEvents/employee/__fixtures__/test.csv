EmployeeID,EmployeeNumber,FirstName,LastName,EmailAddress,MobileNumber
5a7d532f-0b2e-5878-be6f-8ea0d9439150,1,Leroy,Morris,rutuded@ha.sl,(770) 999-2385
3bc9a086-ceac-55b9-9866-af5f54ed435c,2,Bess,Murray,mofizuvul@kuep.dz,(227) 817-7422
06cfa0c0-c903-508c-8b57-64cd89bf42a5,3,Cecelia,Potter,vubpo@ku.cv,(822) 565-4905
afd0eb9d-ecea-59fa-9c7c-630d64fbb119,4,Matthew,Carter,co@la.is,(488) 762-8430
ce80b3e6-2b8d-50ee-b11d-bc5587ba6eea,5,Jason,Henderson,gow@eda.wf,(332) 390-1856
e188c6fd-88e2-5540-a2b3-29be96770299,6,Max,Pearson,uhka@ku.zm,(924) 583-6475
ba9bfd41-6a4f-5e6f-ab31-17079839b76e,7,Herbert,Bennett,henzokib@vezhaem.tk,(715) 883-6422
6fb6ccaf-5c83-5b0e-a845-611eff01f323,8,Wayne,Gutierrez,veogwit@esega.as,(741) 819-8189
330bd4c0-8a64-5cab-b535-462fb86c9df2,9,Leo,Young,atacan@vuvlujjis.as,(780) 382-6041
6833bbe6-bd62-5794-83a6-af927df8ef72,10,Warren,Thomas,voluw@mowhigi.ca,(378) 984-9140
193052ac-7616-5b47-8dd9-aede330af3b9,11,Ida,McDonald,ugno@leofwi.gf,(480) 837-6365
e319fef2-2558-5fd9-9959-cd0aceb5219e,12,Gussie,McCarthy,bo@pe.gt,(920) 712-5143
c14158ba-33cf-553a-8203-5e5e88ef6263,13,Augusta,McCarthy,bazluc@cipigfu.ht,(351) 241-4913
a0cc038c-497b-57b9-955d-8fefe2fecd04,14,Lenora,Hudson,miwbow@cigfin.net,(581) 285-9878
7ad7753e-5011-5dd9-8361-89e659938911,15,Frank,Reid,nibawur@vilegsiv.ss,(267) 988-2304
217f884c-55e2-5ba4-b1db-6574bf7a1387,16,Ophelia,Little,rarudofuk@ni.lb,(651) 937-6410
