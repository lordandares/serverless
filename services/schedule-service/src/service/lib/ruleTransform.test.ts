jest.mock('uuid/v4', () => () => 'action-uuid-1234')

import { schemas } from '@lighthouse/serverless-common'
import * as MockDate from 'mockdate'

import { rulePatternDocument, rulePayload } from '../../__fixtures__'
import { payloadToRulePatternDocument } from './ruleTransform'

beforeEach(() => MockDate.set('2019-10-13T01:00:00.000Z'))
afterEach(() => MockDate.reset())

test('transforms new payload to rulePattern document', async () => {
  const actionId = 'action-id-22222'
  const actor = {
    id: 'actor-id-2222',
    label: 'actor-label-22222',
    type: 'actor-type-22222',
  }

  const { applicationId, locationId, occurrenceId, pk, sk, type } = rulePayload

  const result: schemas.RulePatternDocumentSchema = payloadToRulePatternDocument(
    {
      actor,
      existingDocument: undefined,
      payload: {
        applicationId,
        locationId,
        occurrenceId,
        pk,
        sk,
        type,
      },
    },
  )
  expect(result).toMatchSnapshot()
})

test('transforms existing rulePattern document payload to rulePattern document', async () => {
  const actionId = 'action-id-22222'
  const actor = {
    id: 'actor-id-2222',
    label: 'actor-label-22222',
    type: 'actor-type-22222',
  }
  const existingDocument = rulePatternDocument
  const { applicationId, locationId, occurrenceId, pk, sk, type } = rulePayload

  const result: schemas.RulePatternDocumentSchema = payloadToRulePatternDocument(
    {
      actor,
      existingDocument,
      payload: {
        applicationId,
        locationId,
        occurrenceId,
        pk,
        sk,
        type,
      },
    },
  )
  expect(result).toMatchSnapshot()
})
