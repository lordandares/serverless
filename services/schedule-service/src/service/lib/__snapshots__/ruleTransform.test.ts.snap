// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transforms existing rulePattern document payload to rulePattern document 1`] = `
Object {
  "createdAt": "2019-06-02T00:00:00.000Z",
  "createdBy": Object {
    "id": "created-by-id-11111",
    "label": "created-by-label-1111",
    "type": "created-by-type-11111",
  },
  "groupType": "rule-pattern",
  "matches": Object {
    "occurrence1": Object {
      "pk": "application1-occurrence",
      "sk": "2019-06-02T00:00:00.000Z-occurrence1",
    },
  },
  "pk": "rule-pattern-application-id-1111-location-id-11111",
  "sk": "visit",
  "updatedAt": "2019-10-13T01:00:00.000Z",
  "updatedBy": Object {
    "id": "actor-id-2222",
    "label": "actor-label-22222",
    "type": "actor-type-22222",
  },
}
`;

exports[`transforms new payload to rulePattern document 1`] = `
Object {
  "createdAt": "2019-10-13T01:00:00.000Z",
  "createdBy": Object {
    "id": "actor-id-2222",
    "label": "actor-label-22222",
    "type": "actor-type-22222",
  },
  "groupType": "rule-pattern",
  "matches": Object {
    "occurrence1": Object {
      "pk": "application1-occurrence",
      "sk": "2019-06-02T00:00:00.000Z-occurrence1",
    },
  },
  "pk": "rule-pattern-application-id-1111-location-id-11111",
  "sk": "visit",
  "updatedAt": "2019-10-13T01:00:00.000Z",
  "updatedBy": Object {
    "id": "actor-id-2222",
    "label": "actor-label-22222",
    "type": "actor-type-22222",
  },
}
`;
